pymeasure/instruments/jobinyvon/jobinyvon270m.py

```diff
@@ -0,0 +1,360 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pyvisa.constants import ControlFlow, Parity, StopBits
+import time
+from pyvisa import VisaIOError
+
+
+def get_steps_returns(steps_str: str):
+    """
+    When asked for the number of steps, the spectrometer returns a string in
+    format "o(number of steps)\r", with this function we get rid of the "o" and
+    the "\r" and we convert the resulting string to an integer.
+    """
+    steps_str = steps_str.lstrip('o')
+    steps_str = steps_str.rstrip('\r')
+    return int(steps_str)
+
+
+def read_int(string: str):
+    """
+    When we expect a numeric answer from the spectrometer, it sometimes sends
+    some more characters in between, and with this function we get rid of those
+    extra characters.
+    """
+    numeric_string = ''.join(char for char in string if char.isdigit())
+    return int(numeric_string)
+
+
+class JY270M(Instrument):
+    """
+    This class represents the Jobin-Yvon 270M Driver.
+    """
+
+    _steps_nm = 32  # Number of motor steps per nm
+    _nm_step = 1 / _steps_nm  # nm per motor step
+    _slit_steps_micron = 0.560  # slit steps per micron
+    _slit_microns_step = 1.0 / _slit_steps_micron  # microns per slit step
+
+    _lambda_max = 1171.68  # Maximum value for the wavelength in nm
+    _max_steps = 37494  # maximum number of steps for the grating motor
+    _max_steps_slit = 1102.36  # do not know what this is for?
+
+    @property
+    def default_timeout(self):
+        """
+        Usual timeout needed for normal use.
+        """
+        return 300
+
+    @property
+    def lambda_0(self):
```

Comment:
why do you use properties instead of just setting `lambda_0 = 1171.68` ?

Reply:
We thought that using properties would be more user friendly and in this way we could call the value of the variable from the test files, without needing to write in the test file the number itself we wanted to test (in this case lambda_0).